buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -440,9 +448,22 @@ protected synchronized PoolThreadCache initialValue() {
 
             Thread current = Thread.currentThread();
             if (useCacheForAllThreads || current instanceof FastThreadLocalThread) {
-                return new PoolThreadCache(
+                final PoolThreadCache cache = new PoolThreadCache(
                         heapArena, directArena, tinyCacheSize, smallCacheSize, normalCacheSize,
                         DEFAULT_MAX_CACHED_BUFFER_CAPACITY, DEFAULT_CACHE_TRIM_INTERVAL);
+
+                if (DEFAULT_CACHE_TRIM_RATE > 0) {
+                    EventExecutor executor = ThreadExecutorMap.currentExecutor();
+                    if (executor != null) {
+                        executor.scheduleAtFixedRate(new Runnable() {
+                            @Override
+                            public void run() {
+                                cache.trim();
```

Comment:
try catch Exception to avoid suppressing subsequent executions?

Reply:
@johnou this should never throw, if it does there is something seriously wrong

Reply:
@normanmaurer but unless the app developer has hooked into java.lang.Thread#setDefaultUncaughtExceptionHandler they are not going to know and the exception will be swallowed..